homeassistant/components/ps4/media_player.py

```diff
@@ -0,0 +1,390 @@
+
+    def idle(self):
+        """Set states for state idle."""
+        self.no_title()
+        self._state = STATE_IDLE
+
+    def state_off(self):
+        """Set states for state off."""
+        self.no_title()
+        self._state = STATE_OFF
+        self.update()
+
+    def state_unknown(self):
+        """Set states for state unknown."""
+        self.no_title()
+        self._state = STATE_UNKNOWN
+        _LOGGER.debug("PS4 could not be reached")
+        self._retry = 0
+        self.update()
+
+    def no_title(self):
+        """Update if there is no title."""
+        self._media_title = None
+        self._media_content_id = None
+        self._source = None
+
+    def get_title_data(self, titleid, name):
+        """Get PS Store Data."""
+        app_name = None
+        art = None
+        try:
+            app_name, art = self._ps4.get_ps_store_data(
+                name, titleid, self._region)
+        except TypeError:
+            _LOGGER.error(
+                "Could not find data in region: %s for PS ID: %s",
+                self._region, titleid)
+        finally:
+            self._media_title = app_name or name
+            self._source = self._media_title
+            self._media_image = art
+            self.update_list()
+
+    def update_list(self):
+        """Update Game List, Correct data if different."""
+        if self._media_content_id in self._games:
+            store = self._games[self._media_content_id]
+            if store != self._media_title:
+                self._games.pop(self._media_content_id)
+        if self._media_content_id not in self._games:
+            self.add_games(self._media_content_id, self._media_title)
+            self._games = self.load_games()
+        self._source_list = list(sorted(self._games.values()))
+
+    def load_games(self):
+        """Load games for sources."""
+        g_file = self._games_filename
+        try:
+            games = load_json(g_file)
+            return games
+        except FileNotFoundError:
+            games = {}
+            self.save_games(games)
+            self.load_games()
+
+    def save_games(self, games):
+        """Save games to file."""
+        g_file = self._games_filename
+        try:
+            save_json(g_file, games)
+        except OSError as error:
+            _LOGGER.error("Could not save game list, %s", error)
+
+    def add_games(self, titleid, app_name):
+        """Add games to list."""
+        games = self._games
+        if titleid is not None and titleid not in games:
+            game = {titleid: app_name}
+            games.update(game)
+            self.save_games(games)
+
+    def get_device_info(self, config, status):
+        """Return device info for registry."""
+        _sw_version = status['system-version']
+        _sw_version = _sw_version[1:4]
+        sw_version = _sw_version[0] + '.' + _sw_version[1:]
+        self._info = {
+            'config_entry_id': config.entry_id,
```

Comment:
An entity doesn't provide config entry id. Look how it works for other entities

Reply:
Oops. Didn't know why I put that there.